### ISSUE TITLE
swagger2openapi: Spontaneous generation of invalid components when using x- vendor extensions with $ref

### ISSUE THREAD

**Issue by nulltoken** [Mermade/oas-kit#186]
### Checklist

- [X] Conversion: I have checked my source definition is valid OpenAPI 2.0
- [X] Conversion: On valid OpenAPI 2.0 input, the conversion looks wrong

### Detailed Description

- `x-` vendor extensions located at the root don't look like being properly copied over
- some components are generated. Those components are invalid (from an OpenAPI standpoint).

#### Other stuff

`swagger2openapi:` 5.3.0

**Given the following input:**
```
swagger: '2.0'
basePath: /an/api
host: example.com
schemes:
  - https
info:
  version: '1.0'
  title: 'Root vendor extensions'
  description: 'Should be copied over without generating additional schemas'
x-something:
  one:
    identifier: "1"
    url: "https://one.com"
  another:
    identifier: "Not1"
    url: "https://another.com"
x-something-groups:
  first:
    - color: green
      default:
        backend:
          $ref: "#/x-something/one"
      options:
        - backend:
            $ref: "#/x-something/another"
          criteria:
            - in: "header"
              name: "x-value-id"
              value: [42]
    - color: blue
      default:
        backend:
          $ref: "#/x-something/one"
      options:
        - backend:
            $ref: "#/x-something/another"
          criteria:
            - in: "header"
              name: "x-value-id"
              value: [17]
paths:
  /v1/check:
    get:
      operationId: one
      summary: 'Gets check'
      description: 'Cf. summary.'
      x-backends:
        $ref: "#/x-something-groups/first"
      consumes:
        - application/json
      responses:
        '200':
          description: 'Successful request.'
        '400':
          description: 'Bad request.'
```

**swagger2openapi generates:**
```
openapi: 3.0.0
info:
  version: "1.0"
  title: Root vendor extensions
  description: Should be copied over without generating additional schemas
x-something:
  one:
    $ref: "#/components/schemas/schema1"
  another:
    $ref: "#/components/schemas/schema2"
x-something-groups:
  first:
    $ref: "#/components/schemas/schema3"
paths:
  /v1/check:
    get:
      operationId: one
      summary: Gets check
      description: Cf. summary.
      x-backends:
        $ref: "#/components/schemas/schema3"
      responses:
        "200":
          description: Successful request.
        "400":
          description: Bad request.
servers:
  - url: https://example.com/an/api
components:
  schemas:
    schema1:
      identifier: "1"
      url: https://one.com
    schema2:
      identifier: Not1
      url: https://another.com
    schema3:
      - color: green
        default:
          backend:
            $ref: "#/components/schemas/schema1"
        options:
          - backend:
              $ref: "#/components/schemas/schema2"
            criteria:
              - in: header
                name: x-value-id
                value:
                  - 42
      - color: blue
        default:
          backend:
            $ref: "#/components/schemas/schema1"
        options:
          - backend:
              $ref: "#/components/schemas/schema2"
            criteria:
              - in: header
                name: x-value-id
                value:
                  - 17
```

**Whereas I would hav expected something along the lines of:**
```
openapi: 3.0.0
info:
  version: '1.0'
  title: Root vendor extensions
  description: Should be copied over without generating additional schemas
x-something:
  one:
    identifier: "1"
    url: "https://one.com"
  another:
    identifier: "Not1"
    url: "https://another.com"
x-something-groups:
  first:
    - color: green
      default:
        backend:
          $ref: "#/x-something/one"
      options:
        - backend:
            $ref: "#/x-something/another"
          criteria:
            - in: "header"
              name: "x-value-id"
              value: [42]
    - color: blue
      default:
        backend:
          $ref: "#/x-something/one"
      options:
        - backend:
            $ref: "#/x-something/another"
          criteria:
            - in: "header"
              name: "x-value-id"
              value: [17]
paths:
  "/v1/check":
    get:
      operationId: one
      summary: Gets check
      description: Cf. summary.
      x-backends:
        "$ref": "#/x-something-groups/first"
      responses:
        '200':
          description: Successful request.
        '400':
          description: Bad request.
servers:
- url: https://example.com/an/api
```

**Comment by MikeRalphson** [comment]
Released as part of swagger2openapi@5.3.1

**Comment by nulltoken** [comment]
@MikeRalphson 
![image](https://user-images.githubusercontent.com/92363/61533643-c234c800-aa2d-11e9-924c-3912d1c21c51.png)
